test/testModel.cc

```diff
@@ -776,32 +776,15 @@ SUITE(Canvas)
         godley->table.cell(2,2)="y";
         godley->update();
 
-        
+        // Unit test greatly simplified since flow/stock vars no longer copied via temporary group. For ticket 1039
         unsigned originalNumItems=model->numItems();
-        unsigned originalNumGroups=model->numGroups();
         copyAllFlowVars();
         CHECK_EQUAL(originalNumItems+godley->flowVars().size(),model->numItems());
```

Comment:
This is a good check. But should also check that the relevant items have been inserted into selection, and also that  (I'm guessing in this case) that there are now two copies of the flowVars in model. The way I would approach this is do a loop of model->items, and increment a map<string,int> where the key is the valueId, then afterwards, loop over the flowVars and check that the count is 2. eg

map<string,int> idCnt;
for (auto& i: model->items)
  if (auto v=i.variableCast())
     idCnt[v->valueId()]++;
for (auto v: godley->flowVars())
    CHECK_EQUAL(2, idCnt[v->valueId()])

Reply:
Thank you for this, the test is much more robust now. I will soon commit the new changes.